Python/thread_pthread.h

```diff
@@ -519,3 +519,146 @@ _pythread_pthread_set_stacksize(size_t size)
 }
 
 #define THREAD_SET_STACKSIZE(x) _pythread_pthread_set_stacksize(x)
+
+
+/* pthread native thread-local storage */
+
+#define Py_HAVE_NATIVE_TLS
+/* See thread.c for semantics of the thread-local storage functions */
+
+/* See Python 3's tyhread_pthread.c for similar implementation.
+ * The difference here is that instead of using NATIVE_TSS_KEY_T
+ * (i.e. pthread_key_t) directly, the Python 2 API uses int for keys.
+ * (On many systems, pthread_key_t *is* int, but on some it isn't.)
+ * So, we keep an array mapping these ints to pthread_key_t.
+ */
+
+struct _Py_tss_t {
+    int _is_initialized;
+    pthread_key_t pthread_key;
+};
+
+struct _Py_tss_t *keys = NULL;
+static int next_key = 0;
+static int keys_allocated = 0;
+
+/* Return a new key.  This MUST be called before any other functions in
+ * this family, and callers MUST arrange to serialize calls to this
+ * function.  No violations are detected.
+ *
+ * Note: We avoid re-using keys deleted by PyThread_delete_key.
+ */
+int
+PyThread_create_key(void)
+{
+    unsigned int i;
+    if (next_key == keys_allocated) {
```

Comment:
Why not using pthread_key_create() as in the master branch?

pthread_key_create() prevents the usage of a mutex and so all the mess caused by fork(), like https://bugs.python.org/issue29640